### PR TITLE
Support applying Users and Roles on startup

### PR DESCRIPTION
**Pull request by hugoShaka**
This PR adds support for the kinds `user` and `role` in the `--apply-on-startup` flag.

This is required to bootstrap trust with a bot without user interaction. Seamless tBot bootstrapping is needed for https://github.com/gravitational/teleport/issues/15014.

Changelog: Support applying Users and Roles on startup.

### CODE REVIEW

**Comment by hugoShaka** [comment]
I manually added the reviewers as the bot tried to do (but failed because not everyone is back in the GH org).

**Review by ptgott — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]
TIL about `maps.Keys` - nice!

**Review by r0mant — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

File: lib/auth/init.go [hunk 1]
```diff
@@ -1358,16 +1358,40 @@ func migrateRemoteClusters(ctx context.Context, asrv *Server) error {
...
-// right order.
+// This means consistency is checked. This function support applying resources
+// with dependencies (like a user referring to a role).
 func applyResources(ctx context.Context, service *Services, resources []types.Resource) error {
 	var err error
+	slices.SortFunc(resources, func(a, b types.Resource) int {
+		priorityA, ok := ResourceApplyPriority[a.GetKind()]
+		if !ok {
+			priorityA = 0
+		}
```
[the remark this answers is not in the document]
Comment (hugoShaka) [reply lib/auth/init.go]: Fixed in b527861717e694dce4c3b293147586f3e4066610

**Review by rosstimothy — APPROVED** [review]

**Comment by ptgott** [comment]
@hugoShaka Are there plans to backport this one? Thanks!

**Pull request merged**